disinfect_slam/disinfect_slam.cc

```diff
@@ -0,0 +1,53 @@
+#include "disinfect_slam/disinfect_slam.h"
+
+DISINFSystem::DISINFSystem(
+    std::string camera_config_path,
+    std::string vocab_path,
+    std::string seg_model_path,
+    bool rendering_flag
+) {
+    
+    std::shared_ptr<openvslam::config> cfg = get_and_set_config(camera_config_path);
+    SLAM_ = std::make_shared<SLAMSystem>(cfg, vocab_path);
+    SEG_ = std::make_shared<inference_engine>(seg_model_path);
+    TSDF_ = std::make_shared<TSDFSystem>(0.01, 0.06, 4,
+        get_intrinsics_from_file(camera_config_path), get_extrinsics_from_file(camera_config_path));
+    
+    SLAM_->startup();
+
+    camera_pose_manager_ = std::make_shared<pose_manager>();
+
+    if (rendering_flag) {
+        RENDERER_ = std::make_shared<ImageRenderer>("tsdf", SLAM_, TSDF_, camera_config_path);
+    }
+}
+
+DISINFSystem::~DISINFSystem() {
+    SLAM_->shutdown();
+}
+
+void DISINFSystem::run() {
+    RENDERER_->Run();
+}
+
+void DISINFSystem::feed_rgbd_frame(const cv::Mat & img_rgb, const cv::Mat & img_depth, int64_t timestamp) {
+    cv::Mat my_img_rgb, my_img_depth; // local Mat that will be modified
+    const SE3<float> posecam_P_world = camera_pose_manager_->query_pose(timestamp);
+    cv::resize(img_rgb, my_img_rgb, cv::Size(), .5, .5);
+    cv::resize(img_depth, my_img_depth, cv::Size(), .5, .5);
+    my_img_depth.convertTo(my_img_depth, CV_32FC1, 1. / 4000); // depth scale
```

Comment:
Seems like a lot of hard coded number in a top level interface?

Reply:
fixed depthmap factor